apps/emqx_management/src/emqx_mgmt_data_backup.erl

```diff
@@ -0,0 +1,674 @@
+%% APIs
+%%------------------------------------------------------------------------------
+
+-spec export() -> {ok, backup_file_info()} | {error, _}.
+export() ->
+    export(?DEFAULT_OPTS).
+
+-spec export(map()) -> {ok, backup_file_info()} | {error, _}.
+export(Opts) ->
+    {BackupName, TarDescriptor} = prepare_new_backup(Opts),
+    try
+        do_export(BackupName, TarDescriptor, Opts)
+    catch
+        Class:Reason:Stack ->
+            ?SLOG(error, #{
+                msg => "emqx_data_export_failed",
+                exception => Class,
+                reason => Reason,
+                stacktrace => Stack
+            }),
+            {error, Reason}
+    after
+        %% erl_tar:close/1 raises error if called on an already closed tar
+        catch erl_tar:close(TarDescriptor),
+        file:del_dir_r(BackupName)
+    end.
+
+-spec import(file:filename_all()) ->
+    {ok, #{db_errors => db_error_details(), config_errors => config_error_details()}}
+    | {error, _}.
+import(BackupFileName) ->
+    import(BackupFileName, ?DEFAULT_OPTS).
+
+-spec import(file:filename_all(), map()) ->
+    {ok, #{db_errors => db_error_details(), config_errors => config_error_details()}}
+    | {error, _}.
+import(BackupFileName, Opts) ->
+    case is_import_allowed() of
+        true ->
+            case lookup_file(str(BackupFileName)) of
+                {ok, FilePath} ->
+                    do_import(FilePath, Opts);
+                Err ->
+                    Err
+            end;
+        false ->
+            {error, not_core_node}
+    end.
+
+format_error(not_core_node) ->
+    str(
+        io_lib:format(
+            "backup data import is only allowed on core EMQX nodes, but requested node ~p is not core",
+            [node()]
+        )
+    );
+format_error(ee_to_ce_backup) ->
+    "importing EMQX Enterprise data backup to EMQX is not allowed";
+format_error(missing_backup_meta) ->
+    "invalid backup archive file: missing " ?META_FILENAME;
+format_error(invalid_edition) ->
+    "invalid backup archive content: wrong EMQX edition value in " ?META_FILENAME;
+format_error(invalid_version) ->
+    "invalid backup archive content: wrong EMQX version value in " ?META_FILENAME;
+format_error(bad_archive_dir) ->
+    "invalid backup archive content: all files in the archive must be under <backup name> directory";
+format_error(not_found) ->
+    "backup file not found";
+format_error(bad_backup_name) ->
+    "invalid backup name: file name must have " ?TAR_SUFFIX " extension";
+format_error(Reason) ->
+    Reason.
+
+%%------------------------------------------------------------------------------
+%% Internal functions
+%%------------------------------------------------------------------------------
+
+prepare_new_backup(Opts) ->
+    {{Y, M, D}, {H, MM, S}} = _DateTime = local_datetime(),
+    BackupBaseName = str(
+        io_lib:format("emqx-export-~p-~p-~p-~p-~p-~p", [Y, M, D, H, MM, S])
```

Comment:
Nit: Maybe smth like
```
io_lib:format(
        "~0p.~2..0b.~2..0b.~2..0b.~2..0b.~2..0b.~3..0b",
        [Y, M, D, HH, MM, SS, Ts rem 1000]
    )
```
as in recent Stone's commit to make e.g. names more sorting-friendly?

Reply:
Thanks, great idea!
But I think it's better to keep dashes as separators for date/time, e,g:
```
io_lib:format("emqx-export-~0p-~2..0b-~2..0b-~2..0b-~2..0b-~2..0b.~3..0b",
                      [Y, M, D, HH, MM, SS, Ts rem 1000])
%% emqx-export-2023-06-01-13-06-49.135.tar.gz.
```

Reply:
Fixed